dev-infrastructure/templates/svc-cluster.bicep

```diff
@@ -629,6 +632,22 @@ module svcNSPProfile '../modules/network/nsp-profile.bicep' = {
     associatedResources: [
       svcCluster.outputs.etcKeyVaultId
       rpCosmosDb.outputs.cosmosDbAccountId
+    ]
+    // TODO Add EV2 access here
+    subscriptions: [
+      subscription().id
+    ]
+  }
+}
+
+module svcKVNSPProfile '../modules/network/nsp-profile.bicep' = if (rhDevFixSVCKVAsignNSP) {
```

Comment:
late review: would the condition `resourceGroup().name == serviceKeyVaultResourceGroup` work as well?